src/pages/gateway/hooks.md

```diff
@@ -1,81 +1,70 @@
-  afterAll?: AfterAllTransformObject;
-}
-```
+Hooks increase processing time. Use them sparingly if processing time is important. Hooks are executed in the order you provide them. However, any `blocking` hooks execute before non-blocking hooks.
 
 ## Hook arguments
 
-Hooks accept the following arguments:
+Hooks are plugins that accept the following arguments:
 
-```ts
-interface Hook {
-target: string;
-composer: string;
-blocking: boolean;
+```json
+"hooks": {
+    "beforeAll": {
+        "composer": "<Local or Remote file>",
+        "blocking": true|false
+    }
 }
 ```
 
-- `target` (string) - The target GraphQL node.
+<!-- 
+`target` (string) - The target GraphQL node.
 
     For example, `Query.availableStores` targets [`availableStores`](https://developer.adobe.com/commerce/webapi/graphql/schema/store/queries/available-stores/), which means that if the query calls `availableStores`, then the `composer` will execute.
+-->
 
-- `composer` (string) - The local or remote file location of the function you want to execute when the mesh encounters the node specified by the `target`.
+- `composer` (string) - The local or remote file location of the function you want to execute.
   
     You must add any local scripts to the mesh's [`files` array](../reference/handlers/index.md#reference-local-files-in-handlers). [Local vs remote functions](#local-vs-remote-functions) describes when to use a local or remote function.
 
-    **NOTE**: Local composer functions are limited to 30 seconds. If `blocking` is set to `true` and the function takes longer than 30 seconds, you will receive a `Timeout Error`. In such cases, consider using a [remote composer](#local-vs-remote-functions).
+    **NOTE**: Local composer functions are limited to 30 seconds. If `blocking` is set to `true` and the function takes longer than 30 seconds, you will receive a `Timeout Error`. If you repeatedly encounter this error, consider using a [remote composer](#local-vs-remote-functions).
 
 - `blocking` (boolean) - (`false` by default) Determines if the query waits for a successful return message before continuing the query.
   
-    Only the `before` and `beforeAll` hooks accept the `blocking` argument, which allows you to stop running hooks for a query that does not receive a successful response.
+    The `blocking` argument allows you to stop running hooks for a query that does not receive a successful response.
 
-    If blocking is `true` and the composer returns an error, all future hook executions are canceled and the node's `target` will not be invoked. If multiple objects use the same `target`, an unsuccessful response means that the `target` is not called for the remainder of the operation.
+    If blocking is `true` and the composer returns an error, all future hook executions are canceled.
 
     If blocking is `false` and the composer returns an error, the composer will still be invoked.
 
     Blocking hooks are executed before non-blocking hooks.
```

Comment:
These only are valid in case of hooks that support multiple composers, for instance, `before` and `after` which we havent released yet. 

What do you think about commenting anything that is not pertinent to `beforeAllHooks` for now and enabling it once we have other hooks?

Just to avoidd confusion for the customer